File: specification/specs/x_pack/security/privileges/has_privileges/HasPrivilegesRequest.ts
@rest_spec_name('security.has_privileges')
class HasPrivilegesRequest extends RequestBase {
  path_parts?: {
    user?: Name
  }
  query_parameters?: {}
  body?: {
    application?: ApplicationPrivilegesCheck[]
    cluster?: string[]
    index?: IndexPrivilegesCheck[]
  }
}

File: output/csharp/XPack/Security/Privileges/HasPrivileges/HasPrivilegesRequest.cs
using Nest.XPack;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class HasPrivilegesRequest : RequestBase {
		
		[DataMember(Name="application")]
		public List<ApplicationPrivilegesCheck> Application { get; set; }


		[DataMember(Name="cluster")]
		public List<string> Cluster { get; set; }


		[DataMember(Name="index")]
		public List<IndexPrivilegesCheck> Index { get; set; }

	}
}
